__tests__/integration/one-app.spec.js

```diff
@@ -700,30 +748,72 @@ describe('Tests that require Docker setup', () => {
         });
 
         describe('child module `requiredExternals` invalid usage', () => {
-          const requiredExternalsErrorMatch = /Failed to get external react-intl from root module/;
-
           const moduleName = 'cultured-frankie';
           const version = '0.0.1';
 
-          let requiredExternalsError;
+          afterEach(() => {
+            writeModuleMap(originalModuleMap);
+          });
 
-          beforeAll(async () => {
-            requiredExternalsError = searchForNextLogMatch(requiredExternalsErrorMatch);
+          test('fails to get external `react-intl` for child module as an unsupplied `requiredExternal` - logs failure', async () => {
+            const requiredExternalsErrorMatch = /Failed to get external react-intl from root module/;
+
+            const requiredExternalsError = searchForNextLogMatch(requiredExternalsErrorMatch);
             await addModuleToModuleMap({
               moduleName,
               version,
             });
             // not ideal but need to wait for app to poll;
-
             await waitFor(5000);
-          });
 
-          afterAll(() => {
-            writeModuleMap(originalModuleMap);
+            await expect(requiredExternalsError).resolves.toMatch(requiredExternalsErrorMatch);
           });
 
-          test('fails to get external `react-intl` for child module as an unsupplied `requiredExternal`', async () => {
-            await expect(requiredExternalsError).resolves.toMatch(requiredExternalsErrorMatch);
+          test('fails to get external `react-intl` for child module as an unsupplied `requiredExternal` - Logs reverting message', async () => {
+            const revertErrorMatch = /There was an error loading module (?<moduleName>.*) at (?<url>.*). Reverting back to (?<workingModule>.*)/;
+            const requiredExternalsError = searchForNextLogMatch(revertErrorMatch);
+            await addModuleToModuleMap({
+              moduleName,
+              version,
+            });
+            // not ideal but need to wait for app to poll;
+            await waitFor(5000);
+            const loggedError = await requiredExternalsError;
+            const [,
+              problemModule,
+              problemModuleUrl,
+              workingUrl,
+            ] = revertErrorMatch.exec(loggedError);
+            const gitSha = await retrieveGitSha();
+            await expect(requiredExternalsError).resolves.toMatch(revertErrorMatch);
+            expect(problemModule).toBe('cultured-frankie');
+            expect(problemModuleUrl).toBe(`${testCdnUrl}/${gitSha}/${moduleName}/${version}/${moduleName}.node.js`);
+            // eslint-disable-next-line no-useless-escape
+            expect(workingUrl).toBe(`${testCdnUrl}/${gitSha}/${moduleName}/0.0.0/${moduleName}.node.js\"}`);
+          });
+          test('fails to get external `semver` for child module as an unsupplied `requiredExternal` for new module in mooduleMap', async () => {
```

Comment:
have we not tested the same scenario above when requiring `react-intl`? the only difference I can see is `unhealthy-frank` tries to load a different external `semver`?

If this test is to test the scenario when 2 modules with broken required externals are deployed consecutively, then the afterEach in line 754 is resetting the module map anyway and removing `cultured-frankie` from the previous test

Reply:
This test case test when a new module is added to the module map with an error. The previous test case tests when a module's version is updated with an issue

Reply:
right so we are testing 1 case when the same module is updated (version bump) and it is broken and also when a brand new module which is broken is added for the first time? is that right?